bindings/wallet-jni/src/lib.rs

```diff
@@ -22,14 +22,17 @@ pub unsafe extern "system" fn Java_com_iohk_jormungandrwallet_Wallet_recover(
         .expect("Couldn't get mnemonics String");
 
     let mut wallet: WalletPtr = null_mut();
-    let wallet_ptr: *mut *mut Wallet = &mut wallet;
+    let wallet_ptr: *mut WalletPtr = &mut wallet;
     let result = wallet_recover(&mnemonics_j.to_string_lossy(), null(), 0, wallet_ptr);
 
     let _r = env.release_string_utf_chars(mnemonics, mnemonics_j.as_ptr());
 
-    match result {
-        RecoveringResult::Success => wallet as jlong,
-        _ => 0,
+    if result.is_ok() {
+        wallet as jlong
+    } else {
+        // TODO, the error is completely dismissed here... this is a shame as we could
```

Comment:
We can wrap the error class into a java class maybe?

Reply:
going to throw an exception as a JString instead for now